hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/SelectorOptimizer.java

```diff
@@ -100,10 +100,11 @@ static SelectionKeysSet optimize(Selector selector, TpcLogger logger) {
             logger.finest("Optimized Selector: " + selector.getClass().getName());
             return set;
         } catch (Throwable t) {
-            //System.out.println("Failed to optimize Selector: " + selector.getClass().getName());
-            // we don't want to print at warning level because it could very well be that the target JVM doesn't
-            // support this optimization. That is why we print on finest
-            logger.finest("Failed to optimize Selector: " + selector.getClass().getName(), t);
+            if (logger.isFinestEnabled()) {
```

Comment:
What is the added value for the isFinestEnabled? This code isn't on a hot path.

Reply:
Yeah you are right. At this point it became a habit for me to add if clause for debug level logs.